### ISSUE TITLE
Schema change in parallel with view build can cause bad things to happen

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#11863]
This issue is likely to be a duplicate of #9059 and may be (?) the underlying cause of #11542:

@gleb-cloudius noticed in #11542 a test that does a benign schema change (e.g., even just change the "comment" on the schema) in parallel with view building active, fails in different ways in different releases - crashes when the view building tries to build, broken-looking schemas, etc. I **suspect** that the view build holds an unprotected **schema** pointer instead of a reference-counted **schema_ptr**, so after a schema change the schema gets freed and then used.

Looking at the code I found the following suspicious thing:

```c++
class view_updates final {
    view_ptr _view;
    const view_info& _view_info;
    schema_ptr _base;
    base_info_ptr _base_info;
    std::unordered_map<partition_key, mutation_partition, partition_key::hashing, partition_key::equality> _updates;
    mutable size_t _op_count = 0;
```

This "view_updates" object is used for a potentially long time (across preemptions):  `table::populate_views()` creates a "builder" with `db::view::make_view_update_builder()` and then repeatedly calls `builder.build_some()` with preemptions between those. `make_view_update_builder()` creates an array of these `view_updates` objects.

What I think may be the bug (or at least "a bug") is specifically the `const view_info& _view_info;` above. It has:
```
class view_info final {
    const schema& _schema;
    raw_view_info _raw;
```

This is a bare reference to the schema, not referenced count. I believe it can lead to a use-after-free when used.

**Comment by nyh** [comment]
The `view_info` that I suspect here was added five years ago, in bfb8a3c1725d7ba27ab9dd527d4ba0e66bb8cdd0. So if my suspicion is correct we had this bug for a very long time and it may have caused various bizarre things to happen when changing the schema during view build, including the two issues I referenced above. Now to try to prove this suspicion and fix it...

**Comment by nyh** [comment]
I'm closing this issue and rethink what to do about it, or https://github.com/scylladb/scylladb/issues/9059 or https://github.com/scylladb/scylladb/issues/11542 which may have the same underlying cause.
My proposal above that the view_info is broken is wrong. The thing is, `view_updates` has:
```c++
    const view_info& _view_info;
    schema_ptr _base;
```
So while "_view_info" merely holds an unprotected reference to the schema, the "schema_ptr _base" actually ensures that this same (I think??) schema remains alive.
So there's another cause for this bug. Closing the issue with the incorrect guess for its cause.
